pkg/ccl/sqlproxyccl/server.go

```diff
@@ -11,22 +11,134 @@ package sqlproxyccl
 import (
 	"context"
 	"net"
+	"net/http"
+	"time"
 
+	"github.com/cockroachdb/cockroach/pkg/util/contextutil"
+	"github.com/cockroachdb/cockroach/pkg/util/httputil"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/metric"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
 )
 
+// Server is a TCP server that proxies SQL connections to a
+// configurable backend. It may also run an HTTP server to expose a
+// health check and prometheus metrics.
+type Server struct {
+	mux             *http.ServeMux
+	metrics         *Metrics
+	metricsRegistry *metric.Registry
+
+	promMu             syncutil.Mutex
+	prometheusExporter metric.PrometheusExporter
+}
+
+// NewServer constructs a new proxy server and provisions metrics and health
+// checks as well.
+func NewServer() *Server {
+	mux := http.NewServeMux()
+
+	registry := metric.NewRegistry()
+
+	proxyMetrics := MakeProxyMetrics()
+
+	registry.AddMetricStruct(proxyMetrics)
+
+	s := &Server{
+		mux:             mux,
+		metrics:         &proxyMetrics,
+		metricsRegistry: registry,
+
+		prometheusExporter: metric.MakePrometheusExporter(),
+	}
+
+	mux.HandleFunc("/_status/vars/", s.handleVars)
+	mux.HandleFunc("/_status/healthz/", s.handleHealth)
```

Comment:
Can you explain why it's called "healthz" and not "health"

(also why `/_status` - is that standard?)

Reply:
I was actually just copying the endpoints from CRDB 😅 

Most of cloud's auxiliary services expose metrics at the same endpoint as well.